react/features/virtual-background/components/VirtualBackgroundDialog.js

```diff
@@ -54,23 +59,97 @@ type Props = {
  * @returns {ReactElement}
  */
 function VirtualBackground({ dispatch, t }: Props) {
+    const localImages = jitsiLocalStorage.getItem('virtualBackgrounds');
+    const [ storedImages, setStoredImages ] = useState((localImages && JSON.parse(localImages)) || []);
+    const [ loading, isloading ] = useState(false);
+
+    const toDataURL = url =>
+        fetch(url)
+            .then(response => response.blob())
+            .then(
+                blob =>
+                    new Promise((resolve, reject) => {
+                        const reader = new FileReader();
+
+                        reader.onloadend = () => resolve(reader.result);
+                        reader.onerror = reject;
+                        reader.readAsDataURL(blob);
+                    })
+            );
+
+    const deleteStoredImage = image => {
+        setStoredImages(storedImages.filter(item => item !== image));
+    };
+
+    /**
+     * Updates stored images on local storage.
+     */
+    useEffect(() => {
+        jitsiLocalStorage.setItem('virtualBackgrounds', JSON.stringify(storedImages));
+        if (storedImages.length === 22) {
```

Comment:
Put this constant at the top of the file and leave a comment about why it's 22 please.